adds limus class ship

Ketrai opened the pull request:
## About The Pull Request

![limus](https://user-images.githubusercontent.com/28861083/175167875-99c94f85-fc29-4471-9219-c0ffd90a1529.png)

Xenobiology focused vessel. Alternative to the Metis that leans much more heavily into xenobiology.
- Fully equipped xenobiology for two people.
- Chemlab for making mutagens to change species.
- Janitoral closet to clean up the inevitable mess xenobiology/chemistry will make.
- Trash disposal shute to help get rid of the immense amount of trash xenobiology makes.
- Fooonky shaped ship

## Why It's Good For The Game
This is mainly meant as an alternative to the metis that leans more into xenobiology. Metis is still a far better research ship in comparison. Xenobiology takes a really long time to complete/get going. (1.5-2 hours). And the Metis set up time is around 45 minutes to 1 hour even with multiple people. This cuts down the amount of time you need to get going with xenobiology so you can actually enjoy using the extracts more, or trading with others. (At least on short rounds that don't hit 4 hours.)

Metis also only really has proper space for one xenobiologist, so now more people in the same round can play one.

PROS :
- Xenobiology.
- Xenobio console boards.
- Better parts to get going with xenobiology properly.
- Unique chemlab that makes it a bit easier to make mutagens with xenobiology slime mutagens.

CONS : 
- No autolathe, R&D, cargo, you basically only get xenobio.
- Bad power / engines, demands early mining.
- No materials to build consoles out of.
- No medicine for clone damage. If you get slime glomped, and you're not an IPC, go eat rocks.
- No medical equipment in general, only tools for ghetto surgery.

NOTE : I know xenobiology consoles are buggy right now, as is anything using cameras when the ship moves. The only real fix for this is using a screwdriver to reassemble them. Despite this, I don't feel like it's a good reason to deny players a slime management console. Without it, xenobio is really slow, and kinda tedious. Easily adding 30 minutes to an hour to the role. You still have the option to manually wrangle slimes, but it's really not that fun after 1-2 rounds, unless it's been a while.

## Changelog
:cl:
add: Adds the Limus, a xenobiology focused ship!
/:cl:

Ketrai committed:
adds limus class ship

Zevotech commented:
limus tech tips....

Ketrai committed:
atmos fixes

Ketrai committed:
attempts to fix atmos again

Ketrai committed:
I hate freezer pipes

MusselMain commented:
Why is the RD's Room so cursed?

Calyxman commented:
'No medicine for clone damage. If you get slime glomped, and you're not an IPC, go eat rocks.'
That's.. that's kinda lame
'Bad power / engines, demands early mining'
I feel like people pick specialized ships because they want to get into that specialization, not just mine

Other than that great ship

Latentish commented:
While I think this ship needs some work to buff it out, I do really enjoy this ship. The overall shape is fun and meaningful, it's not horribly overgeared in the departments it's not specialized in, the vast vast majority of the ship is dedicated to the specialization, it has a very good use of space while remaining decently small, and well satisfies its specialization. I also really enjoy that there are two halves of xenobio for the xenobiologists to split, so that it's less of a clusterfuck trying to share the space.

My problems with this are just missing bits and rough edges.
- The slime cage locations and such are good, but the design on their entrances leaves a bit to be desired. There's a size restraint here so there's a bit of a pass. I really wish I could see under the shutters in the preview, but since I can't this might just not be a problem at all. (make sure those windoors can't be opened by slimes).
- The ends of all the disposals pipes into the cages should be disposals exits with reinforced plating underneath. The unattended trunk looks kinda sloppy, adding the proper non-enterable exit won't tremendously impact space.
- Either the main big cage should have a blast door vent to space, or one cage on both sides should have that. Maybe the ones next to the hallways leading north. (This is because it's invaluable to xenobios to have a room that can very quickly be depressurized and cooled to deal with unwanted slimes and other things, as well as the messes they can bring.
- I'd recommend throwing a grey slime core in 4 different cages (two on either side), rather than having one living slime. Core over living is a bit of a preference, but I do feel that more initial starting cores would be good regardless.
- Make sure to check across the other maps with xenobio for some of the missing equipment and QOL. icemoon_surface_slimerancher.dmm is a good reference. (I know you said you have slime monitors, and you have the rest of the general equipment) But having a hyper specialized ship like this means that you can somewhat overequip them for their job. I'd encourage you to come up with some good quality of life contraptions you usually make when you xenobio or like to have. (shower kill chamber with a conveyor to a processor maybe?
- While I think there are a couple ruins and one other ship that use the cake cat, I think that using it takes something away from cooking ships and the like because it's a craftable pet. I also just can't help but feel that a cake cat is a bit inappropriate as a pet for this ship. Maybe make a xeno or one of the aliens you can spawn with xenobio that's passive and named (or just go with the classic rainbow slime or lamar, but that's been done).

Ketrai commented:
> 'No medicine for clone damage. If you get slime glomped, and you're not an IPC, go eat rocks.' That's.. that's kinda lame 'Bad power / engines, demands early mining' I feel like people pick specialized ships because they want to get into that specialization, not just mine
> 
> Other than that great ship

You don't need to mine that much. Just about 10 iron sheets worth for the consoles. And about 30 sheets of plasma to keep you going. This is mostly because I was asked by Quasar to add a reason for people to leave the ship. Being glomped can be avoided easily enough. There's quite a few fire extinguishers too. Otherwise you need puce, which is a crystal plant that can be found on sand and lava planets I believe.

Ketrai commented:
> * The slime cage locations and such are good, but the design on their entrances leaves a bit to be desired. There's a size restraint here so there's a bit of a pass. I really wish I could see under the shutters in the preview, but since I can't this might just not be a problem at all. (make sure those windoors can't be opened by slimes).

![image](https://user-images.githubusercontent.com/28861083/172117815-8d19c7af-9876-4fa4-8b32-a6c49127599b.png)
They have access restriction against slimes.

> * The ends of all the disposals pipes into the cages should be disposals exits with reinforced plating underneath. The unattended trunk looks kinda sloppy, adding the proper non-enterable exit won't tremendously impact space.

This is so monkeys fed into the cages don't end up sitting on the disposal pipe exit where a slime can't get to them.

> * Either the main big cage should have a blast door vent to space, or one cage on both sides should have that. Maybe the ones next to the hallways leading north. (This is because it's invaluable to xenobios to have a room that can very quickly be depressurized and cooled to deal with unwanted slimes and other things, as well as the messes they can bring.

I don't mind adding this for aesthetics, but it's not going to help much gameplay wise. Many mobs from gold slime cores, like statues, don't particularly care about space. (Not to mention statues can push your windows), which means venting their cell could lead players damaging themselves more than anyone. And for slimes you should really be dealing with them by using the extinguisher, rather than venting a whole cell.

> * Make sure to check across the other maps with xenobio for some of the missing equipment and QOL. icemoon_surface_slimerancher.dmm is a good reference. (I know you said you have slime monitors, and you have the rest of the general equipment) But having a hyper specialized ship like this means that you can somewhat overequip them for their job. I'd encourage you to come up with some good quality of life contraptions you usually make when you xenobio or like to have. (shower kill chamber with a conveyor to a processor maybe?

I don't see any equipment I'm missing here other than maybe a water tank you can move. There's biohazard closets which have everything else from bio bags to suits. 5 droppers in total, 3 large beakers per xenobiologist + the two in the grinders. And tier 3 parts just the same like on the slime rancher map. What is it I'm missing here?

![image](https://user-images.githubusercontent.com/28861083/172120212-d4fdf293-419e-49f2-b6d0-1043e00f5b66.png)
The two shutes at the ends of hallways lead to the cold room. 
![image](https://user-images.githubusercontent.com/28861083/172120319-b73fc35a-fa5b-417b-a4a9-e6f3cae7ce93.png)
And the conveyor exit there leads to.
![image](https://user-images.githubusercontent.com/28861083/172120417-7dbee45f-e50a-4dfa-a3ea-996beeb76edc.png)
The processor!

> * While I think there are a couple ruins and one other ship that use the cake cat, I think that using it takes something away from cooking ships and the like because it's a craftable pet. I also just can't help but feel that a cake cat is a bit inappropriate as a pet for this ship. Maybe make a xeno or one of the aliens you can spawn with xenobio that's passive and named (or just go with the classic rainbow slime or lamar, but that's been done).

Well cake cat is one of the cuter animals you can normally randomly get through xenobio. But there is the slime pupper as well. At this point though, one almost has to make new pets just to have more stuff to put on a ship like this. I may see about making something entirely new.

Ketrai committed:
cold room signs & corgi pet

Ketrai commented:
> Why is the RD's Room so cursed?

You know what, you're right. Have a less cursed cubicle.
![image](https://user-images.githubusercontent.com/28861083/172128063-db150c35-84e6-4bb7-aa31-32f97b9fa4a0.png)

Ketrai committed:
RD room update

Ketrai committed:
RD update fix

Ketrai committed:
adds a fennec as a pet

Ketrai commented:
![image](https://user-images.githubusercontent.com/28861083/172239716-da4f980c-7b56-4b95-b761-b6bdb049d86d.png)
![image](https://user-images.githubusercontent.com/28861083/172239785-92261d91-b0e9-420d-b173-a2968bb4f2b0.png)

Not tryna bribe anyone. But fennec. (_Just kidding, if this pr doesn't make it through I'll pr the lil guy separately anyways._)

Ketrai committed:
makes fennec resting a little more natural

Ketrai committed:
Makes sure people can't deconstruct reconstruct the chem dispenser

TypeVar commented:
Extremely nitpicky but 
![image](https://user-images.githubusercontent.com/100787466/172396700-0db8eeb5-9a9c-4a42-80f4-39d4c59c3eb0.png)
(ignore the lizirb) the light tube is attached to the window instead of a wall

Ketrai commented:
> Extremely nitpicky but ![image](https://user-images.githubusercontent.com/100787466/172396700-0db8eeb5-9a9c-4a42-80f4-39d4c59c3eb0.png) (ignore the lizirb) the light tube is attached to the window instead of a wall

I absolutely couldn't find a place to put it where it wouldn't be blocked v-v (As in like, mirror it with the other side since it has a vending machine there.

Ketrai committed:
why is radium in there twice fuk

Yellow-Mushroom commented:
You could use a floorlight instead

Apogee-dev commented on _maps/configs/limus.json:
```diff
@@ -0,0 +1,37 @@
+{
+	"map_name": "Limus-class Xenobiology Cataloguing Vessel",
+	"map_short_name": "Limus-class",
```
Please add a prefix and namelists here. Generic example below:

"prefix": "ISV",
"namelists": ["GENERAL", "SPACE", "NATURAL", "BEASTS"],

Ketrai committed:
namelist

Apogee-dev commented:
All of your fire alarms on the south face of walls are facing the wrong direction, by the way. On a similar note you should make sure the dirs are correct for wall buttons, lightswitches, and intercoms.

The ship could also use a few intercoms in the labs and such so the crew can communicate without SBRs if necessary.

Ketrai committed:
fixes lights et

Ketrai commented:
> All of your fire alarms on the south face of walls are facing the wrong direction, by the way. On a similar note you should make sure the dirs are correct for wall buttons, lightswitches, and intercoms.
> 
> The ship could also use a few intercoms in the labs and such so the crew can communicate without SBRs if necessary.

-  made sure lights are no longer on windows.
- added intercoms everywhere except the bridge, because it's so space starved.
- A couple more little objects such as a labeler, boxes for storage, sandstone in case someone wants to start ghetto botany, some little scenes to make it feel more lived in like mice in the cell, cheese wedges, and mousetraps.
- updated screenshot.
- made sure all buttons are facing the right direction.

Vexylius commented:
![image](https://user-images.githubusercontent.com/22380218/173096450-e364f052-6fb8-47ef-a08a-e6f6783f8584.png)
this on the cockpit looks bad, would pixelshift it as both can be seen at the same time

also could rotate the ORM in the mining place to be like this so it doesn't look weird
![image](https://user-images.githubusercontent.com/22380218/173099216-5f1c337a-fba4-41f1-b13c-f9005b16bd41.png)

would also add blast doors/shutters to the main chamber
![image](https://user-images.githubusercontent.com/22380218/173101316-7b971b50-cb80-485f-9eb9-b73c8d41ce46.png)

Ketrai committed:
fox name & map update

Ketrai committed:
Atmos fixes

Ketrai committed:
machine dir

Ketrai committed:
Blast door for secure pen, fixes

Ketrai commented:
Whole slew of fixes, including requested fixes.

Mainly made the ship a bit bigger. It makes the shape slightly less unique, but now it's got a bit of space for people to make their own tiny departments. Such as a secret retreat for the janitor, and a very basic botany room you can start setting up to do botany. (It just has the board, so you need to find parts!)

This is to give people more to do besides strictly xenobiology.

Cameras are a notable one, you get a couple pre-installed, the rest of the cells have to be set up manually.

One tile wider xenobio because it's cursed to work in otherwise. Now people have space to work around and drop extracts on the floor.

Ketrai committed:
Fixes buttons, lights

Ketrai committed:
more lights

Apogee-dev commented:
Make sure to include shutters/blast doors for the prechargers like you did for the windows. You might want to consider giving the pre-chargers windoor access like most other ships do as well so there's more than just a single directional window separating the thruster bays from space.

Ketrai committed:
faxmachine

Ketrai commented:
> Make sure to include shutters/blast doors for the prechargers like you did for the windows. You might want to consider giving the pre-chargers windoor access like most other ships do as well so there's more than just a single directional window separating the thruster bays from space.

I missed this comment, but yes, makes sense. Will do.

Ketrai committed:
adds engine shutters, basic spare parts.

Latentish commented:
@Apogee-dev could you re-review this one please?

Apogee-dev reviewed (commented):
I did notice that this ship doesn't seem to have a dedicated medical space. While improvised surgical tools would work fine, this ship has no medicine and no way to preserve corpses, and the dispenser in the lab is only a mutagen dispenser. I'd like to see at least one or two basic medkits and some epipens or formaldehyde for that purpose.
Other than that, the aft section of the ship strikes me as really strange, since none of the rooms back there have any clear purpose. I understand wanting to provide construction space, but you may wish to arrange the debris in such a way as to suggest a previous usage. Things like a pile of dirt, old crates, etc. You should also probably make liberal use of dust and dirt decals in these areas, they seem much too clean considering their apparent disuse. (also, consider modifying the x and y coordinates of any trash you place on the ground to make it look less organized and regular. Object positions are somewhat randomized when thrown or dropped in-game, after all, it looks unnatural to have everything exactly in the middle of its tile)

Ketrai committed:
decoration

Ketrai committed:
Removes the console board

Ketrai committed:
scalpel

Ketrai commented:
> I did notice that this ship doesn't seem to have a dedicated medical space. While improvised surgical tools would work fine, this ship has no medicine and no way to preserve corpses, and the dispenser in the lab is only a mutagen dispenser. I'd like to see at least one or two basic medkits and some epipens or formaldehyde for that purpose. Other than that, the aft section of the ship strikes me as really strange, since none of the rooms back there have any clear purpose. I understand wanting to provide construction space, but you may wish to arrange the debris in such a way as to suggest a previous usage. Things like a pile of dirt, old crates, etc. You should also probably make liberal use of dust and dirt decals in these areas, they seem much too clean considering their apparent disuse. (also, consider modifying the x and y coordinates of any trash you place on the ground to make it look less organized and regular. Object positions are somewhat randomized when thrown or dropped in-game, after all, it looks unnatural to have everything exactly in the middle of its tile)

Added a surgery table, a medkit, and a circular saw. (Since crowbar is an awful ghetto tool)

I also personalized the other rooms a bit more. 
- Stage room with costumes, spare parts.
- Unfinished R&D room. Has a server board. I realised a techfab for science spawns in R&D lockers, so I removed  the console board for now. Figured it might be nice to let people research surgeries, but eh, they'll just have to trade for it. With the server they have a partial, useless kit.
- Hydroponics room. Seed vendor, one tray, abandoned crate (as they sometimes have trays). 
- I know the surgery table isn't in an ideal spot, being a bit in the back of the ship and strained for space, but it's the best I could find.

Ketrai committed:
thruster wire

Ketrai committed:
Makes the botany more bearable

Ketrai committed:
windows for slime pens so they're even easier to use.

Ketrai committed:
Limits the ship to one active at a time.

Latentish reviewed (changes requested):
I still don't see a gun anywhere on the ship. Add a simple gun with ammo. A good one to give to a ship like this would be the .38 hunting rifle, a box of its ammo, and the ammo design disk.

Also, these are just annoying, please remove them.
![image](https://user-images.githubusercontent.com/55075747/176536102-3664280a-1f63-46c7-bf3c-b8f9d8d29f91.png)

Latentish commented:
This has been testmerged almost a month now. I'd like to get this closed soon. Please address the pending comments and reviews of this ship. If you're not ready to finish this, you're free to take your time. However we may have to look at un-TMing this soon.

Ketrai commented:
> I still don't see a gun anywhere on the ship. Add a simple gun with ammo. A good one to give to a ship like this would be the .38 hunting rifle, a box of its ammo, and the ammo design disk.
> 
> Also, these are just annoying, please remove them. ![image](https://user-images.githubusercontent.com/55075747/176536102-3664280a-1f63-46c7-bf3c-b8f9d8d29f91.png)

The girders are part of the setup and take like. 2 minutes to remove? I don't really feel like a gun is needed. We've got plenty of more minimal ships without any rifles. And given the power level of xenobio you can get healing to offset the lack of guns and just pickaxe most things to dead until you find better equipment. 

Also I asked if anything still had to be done a couple days ago and no one answered me in discord. I'm on holidays now so any changes will have to wait till after. That being said I've played this ship a lot, haven't found anything to be too bad about it anymore. Can always add a gun as part of an update later.

ZephyrTFA commented:
If you are not going to address the review let by the maptainer assigned to your ship, and are going to actively argue with them about their review; your PR will be closed.

Ketrai commented:
> If you are not going to address the review let by the maptainer assigned to your ship, and are going to actively argue with them about their review; your PR will be closed.

I do feel like I have a little bit of a right to complain? I've been rather cooperative with just about every review thrown my way so far. As well as a radio silence of 7 days between the review before that and then the next when I'm suddenly asked to change things both the other maptainer didn't consider a problem. And to add something I've never been asked to add before. 

But fine. I'll remove the girders, add a gun and reopen this later when I'm back. Would be a shame to see all the work wasted now.

Sylphily commented:
this ship by all means
does not a need a fucking gun, its xeno bio, its already strong, if they are gonna use/want a gun, why not trade for it?
